core/src/macros.rs

```diff
@@ -405,4 +412,10 @@ mod tests {
         },));
         assert_eq!(h2, hlist![true, false, false]);
     }
+
+    #[test]
+    fn unneeded_field_pattern() {
+        #[deny(clippy::unneeded_field_pattern)]
+        let hlist_pat![_, _] = hlist![1, 2];
```

Comment:
Just for completeness, can we have another test that is partial `_`, like the [issue](https://github.com/rust-lang/rust-clippy/issues/11008) @CobaltCause filed?

Reply:
added a test case similar to the issue